sound/soc/sof/intel/hda.c

```diff
@@ -558,7 +558,7 @@ static int check_nhlt_ssp_mclk_mask(struct snd_sof_dev *sdev, int ssp_num)
 	return intel_nhlt_ssp_mclk_mask(nhlt, ssp_num);
 }
 
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC) || IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_SOUNDWIRE)
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_SOUNDWIRE)
```

Comment:
nothing below is SoundWire specific, so it looks like a miss in terms of refactoring?

Reply:
Two functions fixup_tplg_name() and dmic_detect_topology_fixup() are needed by hda_sdw_machine_select() so we have to keep the kernel config check here.